ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -252,9 +254,57 @@ private[kvutils] class TransactionCommitter(
   private def blind: Step =
     (commitContext, transactionEntry) => {
       val blindingInfo = Blinding.blind(transactionEntry.transaction)
-      buildFinalResult(commitContext, transactionEntry, blindingInfo)
+      buildFinalResultState(commitContext, transactionEntry, blindingInfo)
     }
 
+  private def removeUnnecessaryNodes: Step = (commitContext, transactionEntry) => {
+    val transaction = transactionEntry.submission.getTransaction
+    val nodes: Map[String, TransactionOuterClass.Node] =
+      transaction.getNodesList.asScala.map(n => n.getNodeId -> n)(scala.collection.breakOut)
```

Comment:
```suggestion
      transaction.getNodesList.iterator.asScala.map(n => n.getNodeId -> n).toMap
```
I appreciate the intent of not creating intermediate collections, but I would recommend using iterators or views in this regard. The hope is to make it easier on us to migrate to 2.13 (one day), where `CanBuildFrom` no longer exists.